src/ObjLoading/Game/T6/Leaderboard/JsonLeaderboardDefLoader.cpp

```diff
@@ -0,0 +1,152 @@
+
+#include "JsonLeaderboardDefLoader.h"
+
+#include "Game/T6/CommonT6.h"
+#include "Game/T6/Leaderboard/JsonLeaderboardDef.h"
+
+#include <format>
+#include <iostream>
+#include <nlohmann/json.hpp>
+
+using namespace nlohmann;
+using namespace T6;
+
+namespace
+{
+    class JsonLoader
+    {
+    public:
+        JsonLoader(std::istream& stream, MemoryManager& memory)
+            : m_stream(stream),
+              m_memory(memory)
+        {
+        }
+
+        bool Load(LeaderboardDef& leaderboardDef) const
+        {
+            const auto jRoot = json::parse(m_stream);
+            std::string type;
+            unsigned version;
+
+            jRoot.at("_type").get_to(type);
+            jRoot.at("_version").get_to(version);
+
+            if (type != "leaderboard" || version != 1u)
+            {
+                std::cerr << std::format("Tried to load leaderboard \"{}\" but did not find expected type leaderboard of version 1\n", leaderboardDef.name);
+                return false;
+            }
+
+            const auto jLeaderboard = jRoot.get<JsonLeaderboardDef>();
+            return CreateLeaderboardFromJson(jLeaderboard, leaderboardDef);
+        }
+
+    private:
+        static bool CreateTrackTypeFlagsFromJson(const JsonLeaderboardDef& jLeaderboardDef, int& gameFlags)
+        {
+            for (const auto gameFlag : jLeaderboardDef.trackTypes)
+                gameFlags |= gameFlag;
+
+            return true;
+        }
```

Comment:
The [comment from the IW5 PR](https://github.com/Laupetin/OpenAssetTools/pull/206#discussion_r1623209782) applies here as well